src/arvgcport.c

```diff
@@ -248,8 +252,32 @@ arv_gc_port_new (void)
 static void
 arv_gc_port_init (ArvGcPort *gc_port)
 {
+	ArvDomDocument *document;
+	ArvGcRegisterDescriptionNode *register_description;
+	const char *vendor_name;
+	const char *model_name;
+	int i;
+
 	gc_port->priv = arv_gc_port_get_instance_private (gc_port);
 	gc_port->priv->chunk_id = 0;
+	gc_port->priv->has_legacy_infos = FALSE;
+
+	document = arv_dom_node_get_owner_document (ARV_DOM_NODE (gc_port));
```

Comment:
This line fail with error:
> arv_dom_document_get_document_element: assertion 'ARV_IS_DOM_DOCUMENT (self)' failed

I guess that the document is not yet initialized.
Where can I move this code?

In the first call of `_use_legacy_endianness_mechanism`?

Reply:
You can not access the document element in the init function, the node is not yet attached to the XML tree.

This code should be called during the first call of _use_legacy_endianness_mechanism.

Reply:
Ok, I see, thank you.
I have then pushed a new version calling this in the first call of `_use_legacy_endianness_mechanism`.